packages/core/src/operation.ts

```diff
@@ -40,6 +41,10 @@ export interface OperationFunction<TOut> extends Labelled {
   (task: Task<TOut>): Operation<TOut>;
 }
 
+export interface OperationObject<TOut> extends Record<string, unknown> {
```

Comment:
Do we need to include the other valid key types? I don't know. At least it seems ok in the playground https://www.typescriptlang.org/play?ts=4.5.4#code/MYewdgzgLgBCAOBTATgQygS3ALhgVzAwEc9EYIBPAWwCMQAbGAXhgGVq76A6AMxGQAUAcgQp0-IQEoA3AChZUCkhgB5JGkzgAPABUAfMxgDJzAwHFEYMVH5a11rGC2owFPQBod7l27myMYFAoPKjAZPYatvowiAAeQWAAJhAwAEqIoMiJWtDIAQDm7vhgANZgIADuYAYA3rIwDTAA2qIajgC6uBHojrp6cgC+8qCQsLFd6uLIWgBuIBiJBix1jTAAFoj09CC4QhX89IlCRY31jQBULZOaYO3GMDUD7mcNl+y0DFwYQZHIdyaPWQDIA

```suggestion
export interface OperationObject<TOut> extends Record<string|symbol|number, unknown> {
```